src/shared/sam/cli/samCliInit.ts

```diff
@@ -24,11 +32,15 @@ export async function runSamCliInit(initArguments: SamCliInitArgs, context: SamC
         initArguments.runtime,
         '--no-interactive',
         '--app-template',
-        'hello-world',
+        getSamCliTemplateParameter(initArguments.template),
         '--dependency-manager',
         initArguments.dependencyManager
     ]
 
+    if (initArguments.template === eventBridgeStarterAppTemplate) {
+        args.push('--extra-context', JSON.stringify(initArguments.extraContent!))
+    }
```

Comment:
Out of curiosity, what's this special case represent? Will other templates also require something similar? If so, can you break this out into a separate function?

Reply:
In this case extra-context is how we provide the schema information into the template so that the template is updated to refer to teh correct schema, e.g. https://github.com/awslabs/aws-sam-cli-app-templates/blob/master/python3.8/cookiecutter-aws-sam-eventbridge-schema-app-python/%7B%7Bcookiecutter.project_name%7D%7D/template.yaml#L27

but extra-context is a cookiecutter feature that sam exposes via init, so there's nothing preventing using that for other templates in the future

Reply:
`--extra-content` isn't my concern, it's potentially having a ton of different if statements for different templates types; I'd rather not pollute `runSamCliInit` if we could put that logic elsewhere.

Reply:
I'll modify the if statement to check if extraContent is set and push it only in that case. This would prevent polluting the `runSamCliInit` and work for all templates.